### ISSUE TITLE
Adaptive scaling of the learning rate by second order automatic differentiation

### ISSUE THREAD

**Issue by Hiroki11x** [Hiroki11x/Papers#316]
## 一言でいうと
計算のオーダー落としたい

### 論文リンク
https://arxiv.org/abs/2210.14520

## 概要
Deep Neural Networksの最適化の文脈で、我々は自動微分の新しい技法を用いて学習率を再スケールすることを提案する。この技術は、計算複雑度が勾配の計算とヘシアンベクトル積の計算の間にある2次情報である{em curvature}の計算に依存する。(1C,1M)がそれぞれ勾配法の計算時間とメモリフットプリントを表すとすると、新しい手法では全体のコストが(1.5C,2M)または(2C,1M)に増加することになります。このリスケーリングには、自然な解釈ができるという魅力的な特徴があり、これにより、実務家は、パラメータセットの探索とアルゴリズムの収束のどちらかを選択することができます。このリスケーリングは適応的であり、データと降下方向に依存する。数値実験により、異なる探索/収束レジームが強調されている。

## 新規性・差分
我々は、曲率情報を用いて学習率を自動的に再スケーリングする手法を提案し
を提案し、2.2節でこの手法のヒューリスティクスを議論する。この再スケーリングにより
という3つの異なる物理的な領域を選択することができる。
ハイパーエクスプロレーション、エクスプロレーション/コンバージェンストレードオフ、ハイパーコンバージェンス。
- セクション3では、曲率の自動微分とその計算コストについて検討する。
コストについて検討する。
- 第4節では、2.1節で紹介した3つの異なる物理領域に関する議論とともに、数値的なテストを行う。
セクション2.1で紹介した3つの異なる物理領域に関する議論とともに、セクション4で数値テストを行う。

## 実験方法・結果
<img width="548" alt="Screenshot 2022-10-29 at 2 06 29 PM" src="https://user-images.githubusercontent.com/8721858/198833196-18f85244-918a-4f90-bc22-48b17658d9c8.png">
<img width="583" alt="Screenshot 2022-10-29 at 2 06 40 PM" src="https://user-images.githubusercontent.com/8721858/198833201-29a9547f-c264-49e4-bebc-48aa4ffbb7e0.png">
